app/gui/view/graph-editor/src/component/node/input/area.rs

```diff
@@ -205,12 +238,19 @@ impl Model {
         self.widget_tree.set_xy(widgets_origin);
         self.edit_mode_label.set_xy(label_origin);
         self.set_edit_mode(false);
+        self.hide_edit_mode_label();
 
         self
     }
 
     fn set_label_layer(&self, layer: &display::scene::Layer) {
-        self.edit_mode_label.add_to_scene_layer(layer);
+        *self.label_layer.borrow_mut() = layer.downgrade();
+        // We're taking extra measures to ensure that the actual layer of the label will be updated
+        // even if it is already displayed. It should be unnecessary with the current
+        // implementation. See [`Self::show_edit_mode_label`] and [`Self::hide_edit_mode_label`].
```

Comment:
>  It should be unnecessary with the current implementation.

I don't understand this. If it's unnecessary, why do we do this?

Reply:
1. Adam asked a question because the comment in the code is confusing. I see that the code did not change and it is still confusing, so the question was not addressed yet. The comment is also very confusing to me – if it "should be unnecessary with the current implementation", it should not be there.
2. You have full control of if the `set_label_layer` will be called from outside of this component – the method is private, so it will not be called from outside.

I don't think we should include "dead code" that does nothing and is there just in case someone refactors code without thinking about the logic. This makes the whole code hard to understand and follow. In the future, when browsing this code, I would probably spend a lot of time trying to understand why `edit_mode_label_displayed` is needed and why we are managing this state, thinking that probably this solves some issue. If it is not needed, lets not add it :)

Reply:
> You have full control of if the set_label_layer will be called from outside of this component – the method is private, so it will not be called from outside.

This is not true, it is called from the public method of view.

Reply:
Github have fooled me. I actually added a comment yesterday, but it was kept in a pending state, and I didn't notice.

Now it should be visible.

Reply:
Thank you, I like your proposal.